Core/include/Acts/Utilities/TrackHelpers.hpp

```diff
@@ -412,9 +412,10 @@ void calculateTrackQuantities(track_proxy_t track)
 /// @param trimHoles whether to trim holes
 /// @param trimOutliers whether to trim outliers
 /// @param trimMaterial whether to trim pure material states
+/// @param trimOther whether to trim other, non measurement, states
 template <TrackProxyConcept track_proxy_t>
 void trimTrackFront(track_proxy_t track, bool trimHoles, bool trimOutliers,
-                    bool trimMaterial)
+                    bool trimMaterial, bool trimOther)
```

Comment:
```suggestion
                    bool trimMaterial, bool trimNonMeasurement)
```
?
Sounds a bit weird, but might be a bit clearer than *other*.

Reply:
I am also unhappy with the name. `trimNonMeasurement` sounds also a bit like it would to what the other flags do? which might actually be the case right now, I have to double check the logic. I think it would be good if they are all exclusive. So potentially `trimOtherNoneMeasurement`?

Reply:
I guess the problem is that it's not obvious from the signature what the remaining possibilities are.
I'm wondering: couldn't we pass in a `TrackStateType` object that we use as a mask instead of the named arguments?

Reply:
I fear we cannot use `TrackStateType` alone because it is not a standalone object. In parallel we would need a holder for the raw type which would obscure the interface. I think by toggling known flags we would also not be able to model "remove non measurement track states of any remaining type" which is the behavior I would like to achieve here.

Reply:
Ok, you're right. 

About the name: so basically this really only indicates that this will trim all other types of track states that we might add in the future that are not measurements.

Then I guess `trimOtherNoneMeasurement` make the most sense. Should it default to `true`?

Reply:
Ok, let's keep it as is then.